pkg/reconciler/openshift/tektonconfig/common.go

```diff
@@ -103,40 +103,37 @@ func deleteInstallerSet(ctx context.Context, oc versioned.Interface, tc *v1alpha
 // and if installer set which already exist is of older version then it deletes and return false to create a new
 // installer set
 func checkIfInstallerSetExist(ctx context.Context, oc versioned.Interface, relVersion string,
-	tc *v1alpha1.TektonConfig, component string) (bool, error) {
+	tc *v1alpha1.TektonConfig, component string) (*v1alpha1.TektonInstallerSet, error) {
 
 	// Check if installer set is already created
 	compInstallerSet, ok := tc.Status.TektonInstallerSet[component]
-	if !ok {
-		return false, nil
+	if !ok || compInstallerSet == "" {
```

Comment:
do we require this check if the name exist in status?

i understand we create only one for rbac
but as we aim to make logic consistent across all platform and
later move this function in a common pkg
so that we can re use,
we can skip this check and query using label

Reply:
yes, absolutely we can.
in this patch i was focusing on fixing the bug without changing anything around it.

At present, the name of the rbac-istallerset does not change (name: "rbac-resources") so the risk of duplication is less.

Reply:
In a future refactoring we need make the installerset(s) create and manage all rbac related things. This would have been a huge change for next release in the given time frame.